packages/api/src/routes/user.ts

```diff
@@ -302,4 +311,45 @@ router.get("/connect", async (req: Request, res: Response) => {
   return res.status(status.OK).json(users);
 });
 
+/**
+ * Removes the device from the user's profile.
+ */
+async function removeDevice(req: Request, userId: string) {
+  const cxId = getCxIdOrFail(req);
+  const connectedUser = await getConnectedUserOrFail({ id: userId, cxId });
+  const provider = getFromQuery("provider", req);
+  const deviceId = getFromQuery("deviceId", req);
+
+  if (!provider || !deviceId) throw new BadRequestError();
+
+  if (provider === PROVIDER_TENOVI) {
+    const tenovi = new Tenovi();
+    await tenovi.disconnectDevice(connectedUser, String(deviceId));
+  } else {
+    throw new BadRequestError(`Provider not supported: ${provider}`);
+  }
+}
+
+/** ---------------------------------------------------------------------------------------
+* DELETE /user/:userId/device
+*
+* Removes the specified device from the user's profile.
+*
+* @param   {string}  req.params.userId     The internal user ID.
+* @param   {string}  req.query.provider    The device provider.
+* @param   {string}  req.query.deviceId    The device ID to disconnect.
+
+* @return  {{success: boolean}}      If successfully removed.
+*/
+router.delete(
+  "/:userId/device",
+  asyncHandler(async (req: Request, res: Response) => {
+    const userId = getUserIdFrom("params", req).orFail();
+    await removeDevice(req, userId);
+    return res
+      .status(status.OK)
+      .json({ message: `Device ${req.query.deviceId} has been removed for user ${userId}.` });
```

Comment:
We're defining the `deviceId` query param in two places (inside `removeDevice` and here. I suggest we keep the params being defined/retrieved on the route itself.